lsp/nls/src/cache.rs

```diff
@@ -22,15 +22,7 @@ pub trait CacheExt {
 
 impl CacheExt for Cache {
     fn update_content(&mut self, path: impl Into<OsString>, source: String) -> io::Result<FileId> {
-        let path: OsString = path.into();
-        if let Some(file_id) = self.id_of(path.clone()) {
-            self.files_mut().update(file_id, source);
-            // invalidate cache so the file gets parsed again
-            self.terms_mut().remove(&file_id);
-            Ok(file_id)
-        } else {
-            Ok(self.add_string(path, source))
-        }
+        Ok(self.add_tmp(path, source))
```

Comment:
If it's just an alias to `add_tmp`, we can just get rid of `update_content` and use `add_tmp` directly instead. Maybe we can also rename `add_tmp` to `add_in_memory` or something like that, because I reckon that using a method named `add_tmp` inside the LSP to update a document might be a bit confusing.

Reply:
Ok, I went with renaming to `replace_string`, because it made more sense next to `add_string` (which is also in memory). I've removed the `CacheExt` method, but I'm worried that nls needs to do some more work to handle invalidation properly, for example if you modify a file and then query something in a file that imports it. I'll be looking into this more later...